src/lib/containers/widgets/RangeSlider.tsx

```diff
@@ -0,0 +1,225 @@
+import * as React from 'react'
+import './RangeSlider.scss'
+
+import { Slider, Rail, Handles, Tracks, Ticks } from 'react-compound-slider'
+import {
+  SliderItem,
+  GetHandleProps,
+  GetTrackProps,
+} from 'react-compound-slider'
+import { RangeValues } from './Range'
+import { useState } from 'react'
+
+
+/**************** RANGE SLIDER SUBCOMPONENTS *********************/
+
+/***  handles ***/
+
+interface IHandleProps {
+  domain: number[]
+  handle: SliderItem
+  getHandleProps: GetHandleProps
+}
+
+export const Handle: React.FunctionComponent<IHandleProps> = ({
+  domain: [min, max],
+  handle: { id, value, percent },
+  getHandleProps,
+}) => (
+  <div
+    role="slider"
+    className="rangeSlider__handle"
+    aria-valuemin={min}
+    aria-valuemax={max}
+    aria-valuenow={value}
+    style={{
+      left: `${percent}%`,
+    }}
+    {...getHandleProps(id)}
+  />
+)
+
+
+/***   track  ***/
+interface ITrackProps {
+  source: SliderItem
+  target: SliderItem
+  getTrackProps: GetTrackProps
+}
+
+export const Track: React.FunctionComponent<ITrackProps> = ({
+  source,
+  target,
+  getTrackProps,
+}) => (
+  <div
+    className="rangeSlider__track"
+    style={{
+      left: `${source.percent}%`,
+      width: `${target.percent - source.percent}%`,
+    }}
+    {...getTrackProps()}
+  />
+)
+
+/***   tick  ***/
+interface ITickProps {
+  key: string
+  tick: SliderItem
+  count: number
+  mode?: 'SHOWALL' | 'SHOWNONE' | 'SHOWMINMAX'
+}
+
+export const Tick: React.FunctionComponent<ITickProps> = ({
+  tick,
+  count,
+  mode = 'SHOWMINMAX',
+}) => (
+  <div>
+    <div
+      className="rangeSlider__tick"
+      style={{
+        left: `${tick.percent}%`,
+      }}
+    />
+    <div
+      className="rangeSlider__tickInner"
+      style={{
+        marginLeft: `${-(100 / count) / 2}%`,
```

Comment:
Could this margin be positive and go from the right instead? e.g. marginRight: `${(100 / count) / 2}%`

Reply:
I think we want to align the lefts vs the rights though

Reply:
I would think it would act the same perhaps? but its not a big deal either way, ill resolve this,

Reply:
Intuitively I agree. I think the problem is the apply button is a part of the general width.  I could probably refactor the layout and use left margin but since it's not a big deal I won't:)
![image](https://user-images.githubusercontent.com/5190623/72080238-49624a80-32b1-11ea-86a6-3579b447394d.png)